content/blog/2024-06-19-gsoc-environment-variables.mdx

```diff
@@ -0,0 +1,55 @@
+---
+title: "GSoC'24: Supporting User-provided, Long-lived Environmental, Variables for Unikraft Builds"
+description: |
+  The goal of this project is to improve Unikraft's tooling, more especially, the `kraft` command-line too, so that it can dynamically inject user-supplied variables into the build system. 
+publishedDate: 2024-06-19
+tags:
+- GSoC'24   
+- Unikernels
+- KraftKit
+- Golang
+authors:
+- Ujjwal Mahar
+---
+
+## GSoC'24: Supporting User-provided, Long-lived Environmental, Variables for Unikraft Builds
+
+<img width="100px" src="https://summerofcode.withgoogle.com/assets/media/gsoc-generic-badge.svg" align="right" />
+
+
+### Project Overview
+
+Unikraft is a highly adaptable cloud-based library operating system with a high degree of modularity.
+However, developers' ability to add custom variables to the build process is hampered by the rigidity of the current tooling.
+The goal of this project is to improve Unikraft's tooling—more especially, the Kraft command-line tool—so that it can dynamically inject user-supplied variables into the build system.
+Features like changing GCC versions and compile-time settings are included in this.
+Developers can customize Unikraft builds to meet specific needs by providing dynamic configuration, enhancing flexibility.
+
+Impact of this project on the Unikraft Ecosystem:
+- Tooling Framework Improvement
+- Integration with KraftKit Configuration System
+- Modification of Build Invocation Mechanism
+- Default Configuration Setting for CC
+- Kraftfile Integration
+- Dynamic Injection Using Flags in the Command Line
+
+### Progress
+
+Over the past few weeks, I have been following the instructions given by my mentors in the issue [Steps](https://github.com/unikraft/kraftkit/issues/673)
+I have completed two of the steps which include:
+
+1. **Test Building an Application via Make Setup and Build:**
+  Configured the development environment and successfully built a sample Unikraft application using Make.
+
+2. **Exploring Different Toolchain Options Toolchain Testing:**
+  I have experimented with various GCC versions and compile-time settings by passing different options to the make command, ensuring compatibility and robustness.
+
```

Comment:
I would add a short section here explaining the reasoning behind going through this process manually.
Why did you test this? Are you more familiar with the build system? How do you think this helps with the project?